gatorgrade/main.py

```diff
@@ -0,0 +1,35 @@
+"""Use Typer to run gatorgrade to run the checks and generate the yml file."""
+from pathlib import Path
+import typer
+
+app = typer.Typer(add_completion=False)
+FILE = "gatorgrade.yml"
+
+
+@app.callback(invoke_without_command=True)
+def gatorgrade(
+    filename: Path = typer.Option(
+        f"{FILE}", "--config", "-c", help="Name of the yml file."
+    )
+):
+    """Run the GatorGrader checks in the gatorgrade.yml file."""
+    if filename.contains(".yml"):
+        pass
```

Comment:
Marked as a callback, this function will always run for every command, even when the `generate` subcommand is used. This presents a problem since the `gatorgrade` and `gatorgrade generate` commands should do very different things. One way around this is to check the value of the subcommand through the [Typer context](https://typer.tiangolo.com/tutorial/commands/context/#getting-the-context). So, if the value of the invoked subcommand is `None` (which means `generate` was not used), then the behavior of the default `gatorgrade` command should be run. I suggest [adding a script to the `pyproject.toml`](https://python-poetry.org/docs/pyproject/#scripts) to make it easier to run GatorGrade for manually testing the CLI as you make changes. It may be helpful to take a look at [another project's `pyproject.toml`](https://github.com/cmpsc-481-s22-m1/GatorConfig/blob/main/pyproject.toml) for reference.